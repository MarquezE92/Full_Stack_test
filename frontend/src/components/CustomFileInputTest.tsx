import React, { useRef } from 'react';

const CustomFileInputTest = ({ onChange, isOn }) => {
    const ref = useRef(null);
  
    const handleClick = () => {
      ref.current?.click();
    };
  
    return (
      <div className={isOn ? 'custom-file-input-selected' :'custom-file-input'} onClick={handleClick}>
        <input ref={ref} type='file' accept='.csv' onChange={onChange} style={{ display: 'none' }} />
       {isOn ? 'File Selected' : 'Select CSV File'}
      </div>
    );
  };

  export default CustomFileInputTest