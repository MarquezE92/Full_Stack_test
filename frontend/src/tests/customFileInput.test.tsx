import React from 'react';
import TestRenderer from 'react-test-renderer'
import CustomFileInputTest from '../components/CustomFileInputTest';

describe('Custom input testing', ()=> {
    test('Custom input renders correctly', () => {
  // Arrange
  const mockData = {
    isOn: true,
    onChange: jest.fn(), // Mock handleClick function
  };

  
  const treeLine = TestRenderer.create(<CustomFileInputTest {...mockData} />).toJSON()
  expect(treeLine).toMatchSnapshot()
});

})
