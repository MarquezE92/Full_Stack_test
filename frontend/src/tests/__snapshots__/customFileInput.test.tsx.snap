// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonComponent testing ButtonComponent renders correctly 1`] = `
<div
  className="custom-file-input-selected"
  onClick={[Function]}
>
  <input
    accept=".csv"
    onChange={[MockFunction]}
    style={
      {
        "display": "none",
      }
    }
    type="file"
  />
  File Selected
</div>
`;

exports[`Custom input testing Custom input renders correctly 1`] = `
<div
  className="custom-file-input-selected"
  onClick={[Function]}
>
  <input
    accept=".csv"
    onChange={[MockFunction]}
    style={
      {
        "display": "none",
      }
    }
    type="file"
  />
  File Selected
</div>
`;
